src/Build/Logging/BinaryLogger/ExtendedDataFields.cs

```diff
@@ -0,0 +1,14 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+#nullable disable
```

Comment:
Can we define it with nullables?
Then we e.g. could use `ReadString` vs `ReadOptionalString` for the `ExtendedType`

Reply:
Also changed to immutable (removed property setters - what were I thinking :-)